ts/src/profiler.ts

```diff
@@ -126,6 +137,48 @@ async function profileBytes(p: perftools.profiles.IProfile): Promise<string> {
 }
 
 /**
+ * Error which also indicates how long backoff should be in response to the
+ * error.
+ */
+class ResponseError extends Error {
+  backoffMillis?: number;
+  statusCode: number;
+  constructor(response: http.ServerResponse) {
+    let message: number|string = response.statusCode;
+    // tslint:disable-next-line: no-any
+    if ((response as any).statusMessage) {
+      message = response.statusMessage;
+    }
+    super(message.toString());
+    this.statusCode = response.statusCode;
+    if (isServerBackoffResponse(response)) {
+      this.backoffMillis = parseDuration(response.body.details.retryDelay);
```

Comment:
The format is '10s', this is parsed correctly.

Reply:
The server-instructed backoff doesn't have to be exactly 10 seconds though.

Reply:
Do you know where I can find how this might be formatted?
I am a bit concerned because parse-duration doesn't recognize 'u' as microseconds (I've checked, and it does parse strings like '5d6h1m52s100ms').